lib/active_model/serializer/adapter/json.rb

```diff
@@ -4,44 +4,130 @@ class ActiveModel::Serializer::Adapter::Json < ActiveModel::Serializer::Adapter
 
         def serializable_hash(options = nil)
           options ||= {}
-          if serializer.respond_to?(:each)
-            result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
-          else
-            hash = {}
 
-            core = cache_check(serializer) do
-              serializer.attributes(options)
+          hash =
+            if serializer.respond_to?(:each)
+              serializable_hash_for_collection(options)
+            else
+              serializable_hash_for_single_resource
```

Comment:
Yup, I'll play around with some things tomorrow and see if, indeed, there are supposed to be options passed in to the single resource. As it is currently, it uses the serializers own options. See line: 130